prog/pwm/fancontrol

```diff
@@ -540,7 +546,12 @@ function UpdateFanSpeeds
 		  then pwmval=$maxpwm # over max temp, use defined max pwm
 		else
 		  # calculate the new value from temperature and settings
-		  pwmval="(${tval}-${mint})*(${maxpwm}-${minso})/(${maxt}-${mint})+${minso}"
+		  if "$fccharcur" = "LIN"
+		  then        # linear transfer function
+		     pwmval=$(( (${tval}-${mint})*(${maxpwm}-${minso}) / (${maxt}-${mint}) + ${minso}))
```

Comment:
This would be better fixed with a separate patch (as bugfix only). 
Also, I just found that ${...} isn't necessary within $(( )). (run shellcheck command).
$(((tval-mint)*(maxpwm-minso) / (maxt-mint) + minso)) should do and looks much nicer.

For the actual calculations, I think it would be better to introduce temporary variables for (${tval}-${mint}), (${maxpwm}-${minso}), and (${maxt}-${mint}) since they are now all used multiple times. This would also make it easier to understand the code.